### PR TITLE
Add the HuggingFace Model Evaluation Notebook to the website

### PR DESCRIPTION
**Pull request by BenWilson2**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/BenWilson2/mlflow/pull/10789?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10789/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10789
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Adds the HuggingFace model evaluation example to the site.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [X] Yes. I've updated:
  - [X] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by BenWilson2** [comment]
@annzhang-db I had to convert the details dropdown div section to base markdown (the details element doesn't render correctly within notebooks on the site) due to an issue with the html parser in sphinx notebook conversion (it wraps new divs in <p> elements, which breaks the <details> lineage). Everything else is just porting over your notebook for site rendering!

File: docs/source/llms/llm-evaluate/notebooks/huggingface-evaluation.ipynb [hunk 1]
```diff
@@ -0,0 +1,1099 @@
...
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Load Evaluation Data\n",
+    "\n",
+    "Load in a dataset from Hugging Face Hub to use for evaluation."
```
Comment (B-Step62) [inline docs/source/llms/llm-evaluate/notebooks/huggingface-evaluation.ipynb:155]: Could you add a quick description for what each column means?

File: docs/source/llms/llm-evaluate/notebooks/huggingface-evaluation.ipynb [hunk 2]
```diff
@@ -0,0 +1,1099 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since we are evaluating how well our model can provide an answer to a given instruction, we may want to choose some metrics to help measure this on top of any builtin metrics that `mlflow.evaluate()` gives us.\n",
+    "\n",
+    "Let's measure how well out model is doing on following two metrics:\n",
+    "- Is the answer correct? Let's use the predefined metric `answer_correctness` here.\n",
```
Comment (B-Step62) [inline docs/source/llms/llm-evaluate/notebooks/huggingface-evaluation.ipynb:340]: nit: not rendered as bullet points properly

File: docs/source/llms/llm-evaluate/notebooks/huggingface-evaluation.ipynb [hunk 3]
```diff
@@ -0,0 +1,1099 @@
...
+    "- Is the answer fluent, clear, and concise? We will define a custom metric `answer_quality` to measure this.\n",
+    "\n",
+    "We will need to pass these into `extra_metrics` argument for `mlflow.evaluate()`.\n",
+    "\n",
+    "#### What is an Evaluation Metric?\n",
+    "\n",
+    "An evaluation metric encapsulates any quantitative or qualitative measure you want to calculate for your model. For each model type, `mlflow.evaluate()` will automatically calculate some set of builtin metrics. Refer [here](https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate) for which builtin metrics will be calculated for each model type. You can also pass in any other metrics you want to calculate as extra metrics. MLflow provides a set of predefined metrics that you can find [here](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html), or you can define your own custom metrics. In the example here, we will use the predefined metric `mlflow.metrics.genai.answer_correctness`.\n",
+    "\n",
+    "##### Custom Metrics\n",
```
Comment (B-Step62) [inline docs/source/llms/llm-evaluate/notebooks/huggingface-evaluation.ipynb:349]: Can we skip describing `make_metric` explanation? Doesn't seem to be related to what we are doing here.

File: docs/source/llms/llm-evaluate/notebooks/huggingface-evaluation.ipynb [hunk 4]
```diff
@@ -0,0 +1,1099 @@
...
+    "\n",
+    "Let's measure how well out model is doing on following two metrics:\n",
+    "- Is the answer correct? Let's use the predefined metric `answer_correctness` here.\n",
+    "- Is the answer fluent, clear, and concise? We will define a custom metric `answer_quality` to measure this.\n",
+    "\n",
+    "We will need to pass these into `extra_metrics` argument for `mlflow.evaluate()`.\n",
+    "\n",
+    "#### What is an Evaluation Metric?\n",
+    "\n",
+    "An evaluation metric encapsulates any quantitative or qualitative measure you want to calculate for your model. For each model type, `mlflow.evaluate()` will automatically calculate some set of builtin metrics. Refer [here](https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate) for which builtin metrics will be calculated for each model type. You can also pass in any other metrics you want to calculate as extra metrics. MLflow provides a set of predefined metrics that you can find [here](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html), or you can define your own custom metrics. In the example here, we will use the predefined metric `mlflow.metrics.genai.answer_correctness`.\n",
```
Comment (B-Step62) [inline docs/source/llms/llm-evaluate/notebooks/huggingface-evaluation.ipynb:347]:
```suggestion
    "An evaluation metric encapsulates any quantitative or qualitative measure you want to calculate for your model. For each model type, `mlflow.evaluate()` will automatically calculate some set of builtin metrics. Refer [here](https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate) for which builtin metrics will be calculated for each model type. You can also pass in any other metrics you want to calculate as extra metrics. MLflow provides a set of predefined metrics that you can find [here](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html), or you can define your own custom metrics. In the example here, we will use the combination of predefined metric `mlflow.metrics.genai.answer_correctness` and custom metric for quality evaluation .\n",
```

**Review by B-Step62 — APPROVED** [review]
Sorry this PR has been left a while in the bottom of my queue.... Great contents, just left a few nit but looks good to me overall!

**Review by chenmoneygithub — APPROVED** [review]
Sorry +1, for some reason this PR did not pop up in my mailbox..

**Review by WeichenXu123 — APPROVED** [review]
LGTM after addressing @B-Step62's comments :)
